deep-map/index.d.ts

```diff
@@ -0,0 +1,21 @@
+import { WritableAtom } from '../atom/index.js'
+import { AllKeys, GetPath } from './path.js'
+
+export * from './path.js'
+
+type Listener<T extends Record<string, unknown>> = (
+  listener: (value: T, changedKey: undefined | AllKeys<T>) => void
+) => () => void
+
+export type DeepMapStore<T extends Record<string, unknown>> = Omit<
+  WritableAtom<T>,
+  'setKey' | 'listen' | 'subscribe'
+> & {
+  setKey: <K extends AllKeys<T>>(key: K, value: GetPath<T, K>) => void
+  listen: Listener<T>
+  subscribe: Listener<T>
+}
+
+export function deepMap<T extends Record<string, unknown>>(
```

Comment:
We need TypeDocs here

Reply:
I like the idea. Some users already asked for something like this.

Reply:
@ai Have a look. Added docs to all exported functions and to README itself.